diff --git a/bin.js b/bin.js
index 7ca3ab5fd0babcfde234427387df4904b9e5f6f4..e2c6341228abc586bd5531b00e416269ce0a5947 100644
--- a/bin.js
+++ b/bin.js
@@ -18,6 +18,7 @@ if (!buildFromSource()) {
 function build () {
   var win32 = os.platform() === 'win32'
   var args = [win32 ? 'node-gyp.cmd' : 'node-gyp', 'rebuild']
+  var shell = win32
 
   try {
     var pkg = require('node-gyp/package.json')
@@ -26,9 +27,10 @@ function build () {
       path.join(require.resolve('node-gyp/package.json'), '..', typeof pkg.bin === 'string' ? pkg.bin : pkg.bin['node-gyp']),
       'rebuild'
     ]
+    shell = false
   } catch (_) {}
 
-  proc.spawn(args[0], args.slice(1), { stdio: 'inherit', shell: win32, windowsHide: true }).on('exit', function (code) {
+  proc.spawn(args[0], args.slice(1), { stdio: 'inherit', shell, windowsHide: true }).on('exit', function (code) {
     if (code || !process.argv[3]) process.exit(code)
     exec(process.argv[3]).on('exit', function (code) {
       process.exit(code)
